core/client/fs/src/main/java/alluxio/client/netty/NettyRPC.java

```diff
@@ -43,9 +45,17 @@ private NettyRPC() {}  // prevent instantiation
   public static ProtoMessage call(final NettyRPCContext context, ProtoMessage request)
       throws IOException {
     Channel channel = Preconditions.checkNotNull(context.getChannel());
-    Promise<ProtoMessage> promise = channel.eventLoop().newPromise();
+    final Promise<ProtoMessage> promise = channel.eventLoop().newPromise();
     channel.pipeline().addLast(new RPCHandler(promise));
-    channel.writeAndFlush(new RPCProtoMessage(request));
+    channel.writeAndFlush(new RPCProtoMessage(request)).addListener(new ChannelFutureListener() {
+      @Override
+      public void operationComplete(ChannelFuture future) throws Exception {
+        if (future.cause() != null) {
+          promise.tryFailure(future.cause());
+          future.channel().close();
+        }
+      }
+    });
```

Comment:
How much do we we save by having this compared to waiting for `RPCHandler`'s `exceptionCaught`?